tests/ert/unit_tests/analysis/test_es_update.py

```diff
@@ -580,19 +585,6 @@ def test_temporary_parameter_storage_with_inactive_fields(
         np.testing.assert_array_equal(ds["values"].values[0], fields[iens]["values"])
 
 
-def test_that_observations_keep_sorting(snake_oil_case_storage, snake_oil_storage):
```

Comment:
Is this superseded by something else?

Reply:
Observations are now always sorted by `observation_key` when they are made, and there is one dataset per response type. This test checks that the list of obs keys from the ert config is the same as the list of obs keys from the experiment. I did add `.observation_keys` to experiment, which manually gets the obs key list across all response type, and felt it would be a bit duplicate to make the same thing in enkfobs only to assert that it corresponds with the order in the experiment. 

The test `test_that_enkf_obs_keys_are_ordered` does assert that all created enkfobs are sorted by observation_key though. 

Currently there is a loop over all obs keys which makes it more sensible to assert this order, but now we know that the dataset itself is always ordered by obs key (asserted by test mentioned above), so the result of the `join` in `_es_update` should also have that order